README.md

```diff
@@ -76,98 +127,91 @@ The standard state format for an input in redux looks like this:
-import { connect, Provider } from 'react-redux';
+import { createInputsReducer, connectWithInputs } from 'redux-inputs';
+import { Provider } from 'react-redux';
 import thunk = from 'redux-thunk';
 
 const inputConfig = {
-	email: {
-		defaultValue: 'test@example.com',
-		validator: (value) => typeof value === 'string' && value.indexOf('@') >= 0
-	}
+    email: {
+        defaultValue: 'test@example.com',
+        validator: (value) => typeof value === 'string' && value.indexOf('@') >= 0
+    }
 };
 const reducer = combineReducers({
     inputs: createInputsReducer(inputConfig)
 });
 const store = createStore(reducer, applyMiddleware(thunk));
 
+let EmailInput = ({id, value, error, onChange}) => (
+    <div>
+        <input name={id} onChange={(e) => onChange(e.target.value)}/>
+        {error ? <div>Your email must contain an @</div> : null} 
+    </div>
+);
+EmailInput = InputsWrapper(EmailInput);
+
 function Form(props) {
-	  const { dispatch, inputs } = props;
-    const inputProps = getInputProps(inputConfig, inputs, dispatch);
-    const isEmailError = inputs.email.error;
+    const { dispatch, inputs, inputProps } = props;
     return (
         <form>
-            <input id="email"
-                   value={inputProps.email.value}
-                   onChange={(e) => {
-                       inputProps.email.dispatchChange({ email: e.target.value })
-                   }} />
-             { isEmailError? <p className="error">Your email must contain an @</p> : null}
-             <h3>Input state</h3>
-             <pre>{JSON.stringify(inputs, null, 2)}</pre>      
+            <EmailInput {...inputProps.email}/>
+            <h3>Input state</h3>
+            <pre>{JSON.stringify(inputs, null, 2)}</pre>      
         </form>
     )
 }
-const FormContainer = connect(s => s)(Form);
-ReactDOM.render(<Provider store={store}><FormContainer /></Provider>, document.getElementById('container'));
+const ConnectedForm = connectWithInputs(inputConfig)(s => s)(Form);
+ReactDOM.render(<Provider store={store}><ConnectedForm /></Provider>, document.getElementById('container'));
 ```
-[Interactive Demo](./examples.html)
 
-The object passed to dispatchChange can have any number of key-value pairs in it.
 
-`dispatchChange` also returns a promise for when async validation runs and you want to handle the result after it's done.
+[Interactive Demo](./examples.html)
 
 With this set up, you are able make changes to inputs and have them declaratively
  validated and state synchronized with your store. Because this could send actions
  to your store every keystroke, you probably want to use something like a
  [BlurInput](http://khan.github.io/react-components/#blur-input).
 
-By keeping these two redux-inputs functions flexible, it is easy to have multiple
- forms by making multiple inputConfigs. You can also make dynamic forms by
- dynamically creating input configs, replacing reducers, and creating dynamic components.
+## Action Creators/Thunks
```

Comment:
I'm considering we start passing these down as props to connectWithInput components in some way, so they are pre-bound to inputConfig and dispatch. The most common use case for using any of these right now is 
```
import { validateInputs } from 'redux-inputs'
...
const { dispatch } = this.props;
dispatch(validateInputs(inputConfig)).then(() => /* submit form */)
```

It could instead be something like
```
const { validateInputs } = this.props;
validateInputs().then(() => /* submit form */)
```